central/delegatedregistryconfig/convert/convert.go

```diff
@@ -0,0 +1,70 @@
+// Package convert provides utility functions for converting between the various
+// DelegatedRegistryConfig types.
+//   - "Storage"     (storage.DelegatedRegistryConfig) - for persistance
+//   - "API"         (v1.DelegatedRegistryConfig)      - for exposed REST/GRPC API
+package convert
+
+import (
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+// StorageToAPI converts a delegated registry config from the type used by storage (db) to
+// the type used by the GRPC/REST API
+func StorageToAPI(from *storage.DelegatedRegistryConfig) *v1.DelegatedRegistryConfig {
+	if from == nil {
+		return nil
+	}
+
+	var regs []*v1.DelegatedRegistryConfig_DelegatedRegistry
+
+	if len(from.Registries) > 0 {
+		regs = make([]*v1.DelegatedRegistryConfig_DelegatedRegistry, len(from.Registries))
+
+		for i, reg := range from.Registries {
+			regs[i] = &v1.DelegatedRegistryConfig_DelegatedRegistry{
+				ClusterId:    reg.ClusterId,
+				RegistryPath: reg.RegistryPath,
+			}
+		}
+	}
+
+	// defaults to 0 (NONE) if not found in map
+	enabledFor := v1.DelegatedRegistryConfig_EnabledFor_value[from.EnabledFor.String()]
```

Comment:
I always opt for manual `switch`, but this works as long as we are careful. This can be dangerous if we were to change the name of the enum (I believe that's allowed and compatible in proto as long as the tag is not changed)

Reply:
I'm thinking let's make a function to do this conversion ourselves for safety

Reply:
Wouldn't changing the name of the enum cause a compile time error? The map would have a different name in that case unless I'm missing something.  What did you have in mind for a function? Each of the types could potentially have a different set of names to values (`int32`'s) in the future, the map seemed to be the safest option in that case.  Unit tests would fail as well if the values changed to something unexpected.

Reply:
I was wondering what may happen if `from.EnabledFor.String()` is no longer something which matches `NONE`, `ALL`, or `SPECIFIC`. For example, another option is added for this proto but not the other. What will this default to? Or, what if we change the API's name but not the storage proto's name? We may default to something unexpected, which we would catch with a compile-time error instead. Let me know if that makes sense

Reply:
> I was wondering what may happen if from.EnabledFor.String() is no longer something which matches NONE, ALL, or SPECIFIC. For example, another option is added for this proto but not the other. What will this default to?

I believe that use case is captured in the [convert_test.go](https://github.com/stackrox/stackrox/blob/dc/ROX-16696-delegated-reg-config-storage-api/central/delegatedregistryconfig/convert/convert_test.go#L56) - For good measure however, the below test shows that an entry not found in the map will default to `0`

```go
func TestTemp(t *testing.T) {
	enabledFor := v1.DelegatedRegistryConfig_EnabledFor_value["HAPPY TUESDAY!"]
	fmt.Println(enabledFor)
}
```
Output:
```
=== RUN   TestTemp
0
--- PASS: TestTemp (0.00s)
```

> Or, what if we change the API's name but not the storage proto's name? We may default to something unexpected, which we would catch with a compile-time error instead. Let me know if that makes sense

If we change one name and not the other the tests in `convert_test.go` will fail.

For good measure changed `proto/api/v1/delegated_registry_config_service.proto` to have:

```proto
...
    enum EnabledFor {
        HELLO     = 0;
        WORLD      = 1;
        YIPPIE = 2;
    }
...
```
And the test case no longer compiles:
![image](https://github.com/stackrox/stackrox/assets/119438707/ebeb8626-2798-44c6-8956-a00c81ab309b)

Reply:
as long as it is captured somewhere by compile time errors, I'm ok with this